### PR TITLE
Fix #4267 by passing `warn` option to package textcomp.

### LINKED ISSUE #4267: PDF build broken by № (Unicode U+2116) character
#### Procedure to reproduce the problem

File `index.rst`

```rst
Welcome
=======

№
```

```
make latexpdf
```

#### Error logs / results
```
! Package textcomp Error: Symbol \textnumero not provided by
(textcomp)                font family ptm in TS1 encoding.
(textcomp)                Default family used instead.

See the textcomp package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.71 №
        
?
```

#### Expected results

Compilation succeeds because after all fall-back font (Computer Modern) contains the glyph contrarily to Times TeX font (`\usepackage{times}` default with Sphinx)

### Environment info
- OS: Mac
- Python version: 3.5.4
- Sphinx version: 1.6.3, current stable
- TeXLive 2017

### How to fix it?

Notice that ``See the textcomp package documentation`` hint is hard to follow because as far as I know one finds the documentation in the comments of the LaTeX source code, and in their PDF build, `source2e.pdf` (it might also be in some non-free book like the LaTeX companion, not checked). But doing this I found that the package `textcomp` admits the `warn` option which in such case turns the Error (stopping the build) into a simple LaTeX warning (which we can ignore).

So I will make PR for `sphinx.sty` to use this `warn` option.

### PR DESCRIPTION
**Pull request by jfbu**
This silences its (i.e. LaTeX package textcomp) silly build-breaking errors in case some font substitution proved necessary for the characters it supports.

I consider this bugfix, hence based it on stable.

### Relates

- #4267 
- https://github.com/python/cpython/pull/4069

### CODE REVIEW

**Review by tk0miya — APPROVED** [review]

**Comment by jfbu** [comment]
Thanks for review, @tk0miya ! Merging...
